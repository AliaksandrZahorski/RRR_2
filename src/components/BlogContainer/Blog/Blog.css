.blog {
  background: #e1e3de;
  max-width: 800px;
  padding: 0 0 0 0;
  margin: 0px auto;
}
