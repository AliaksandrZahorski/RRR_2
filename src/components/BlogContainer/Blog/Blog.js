import React from 'react';
import { connect } from 'react-redux';

import { deleteRecord } from '../../../actions';

const mapDispatchToProps = dispatch => ({
  onDeleteRecord(id) {
    dispatch(deleteRecord(id));
  },
});

const Blog = ({ data, onDeleteRecord }) => {
  const deleteRecord = id => onDeleteRecord(id);
  return(
    <div className="blog">
      <p><strong>{data.title}</strong></p>
      <p>{data.author}</p>
      <p>{data.text}</p>
      <button
        type="submit"
        onClick={e => {
          e.preventDefault();
          deleteRecord(data._id);
        }}
      >
        Delete
      </button>
    </div>
  );
}

export default connect(null, mapDispatchToProps)(Blog);
