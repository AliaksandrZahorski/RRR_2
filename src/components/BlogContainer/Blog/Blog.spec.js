import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Blog from './Blog';

const mockStore = configureMockStore([thunk]);

describe('Table', () => {
  it('render correctly', () => {
    const storeStateMock = {
    };
    const data = {
    };
    const store = mockStore(storeStateMock);
    const component = shallow(
      <Blog
        store={store}
        data={data}
      />
    ).shallow();
    expect(component).toMatchSnapshot();
  });
}); 