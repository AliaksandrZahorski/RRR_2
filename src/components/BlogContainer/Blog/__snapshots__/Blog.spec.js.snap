// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Blog
    data={Object {}}
    onDeleteRecord={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <strong />
        </p>,
        <p />,
        <p />,
        <button
          onClick={[Function]}
          type="submit"
        >
          Delete
        </button>,
      ],
      "className": "blog",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "strong",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <strong />
          </p>,
          <p />,
          <p />,
          <button
            onClick={[Function]}
            type="submit"
          >
            Delete
          </button>,
        ],
        "className": "blog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "strong",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
